packages/electron-commons/src/node-entry.ts

```diff
@@ -1,78 +1,112 @@
+        getRuntimeArguments: () => command.runOptions,
+    });
 
-        const metadataProviderHost = new BaseHost();
-        metadataProviderHost.name = METADATA_PROVIDER_ENV_ID;
-        rootCom.registerEnv(METADATA_PROVIDER_ENV_ID, metadataProviderHost);
+    const metadataProviderHost = new BaseHost();
+    metadataProviderHost.name = METADATA_PROVIDER_ENV_ID;
+    rootCom.registerEnv(METADATA_PROVIDER_ENV_ID, metadataProviderHost);
 
-        config.push(
-            COM.use({
-                config: {
-                    connectedEnvironments: {
-                        [ENGINE_ROOT_ENVIRONMENT_ID]: {
-                            id: ENGINE_ROOT_ENVIRONMENT_ID,
-                            host: rootEngineEnvHost,
-                        },
-                        [METADATA_PROVIDER_ENV_ID]: {
-                            id: METADATA_PROVIDER_ENV_ID,
-                            host: metadataProviderHost,
-                        },
+    config.push(
+        COM.use({
+            config: {
+                connectedEnvironments: {
+                    [ENGINE_ROOT_ENVIRONMENT_ID]: {
+                        id: ENGINE_ROOT_ENVIRONMENT_ID,
+                        host: rootEngineEnvHost,
+                    },
+                    [METADATA_PROVIDER_ENV_ID]: {
+                        id: METADATA_PROVIDER_ENV_ID,
+                        host: metadataProviderHost,
                     },
                 },
-            })
-        );
+            },
+        })
+    );
+
+    const { close } = await runIPCEnvironment({
+        type: 'node',
+        name: environmentName,
+        bundlePath,
+        childEnvName: environmentContextName,
+        featureName,
+        config,
+        features,
+        options: runtimeOptions,
+        context: basePath,
+        parentEnvName,
+        env,
+    });
+
+    disposeEnv = close;
+}
+
+async function handleDisposeCommand() {
+    await Promise.all([disposeEnv?.()]);
+}
+
+const messageHandler = (message: unknown) => {
+    const command = message as NodeEnvironmentCommand;
+
+    switch (command.id) {
+        case 'nodeEnvironmentStartupCommand':
+            handleStartupCommand(command).catch((e) => {
+                emitEvent<NodeEnvironmentEvent>(process, {
+                    id: 'nodeEnvironmentInitFailedEvent',
+                    error: toError(e).message,
+                });
+            });
+            break;
 
-        await runIPCEnvironment({
-            type: 'node',
-            name: environmentName,
-            bundlePath,
-            childEnvName: environmentContextName,
-            featureName,
-            config,
-            features,
-            options: runtimeOptions,
-            context: basePath,
-            parentEnvName,
-            env,
-        });
+        case 'nodeEnvironmentDisposeCommand':
+            handleDisposeCommand()
+                .then(() => process.off('message', messageHandler))
```

Comment:
why isn't this inside handleDisposeCommand?

Reply:
fixed